spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/types/external.scala

```diff
@@ -287,99 +237,22 @@ object external {
-    tableDefinition.comment.foreach(tblEntity.setAttribute("comment", _))
-    tableDefinition.viewText.foreach(tblEntity.setAttribute("viewOriginalText", _))
-
-    tblEntity.setRelationshipAttribute("db", AtlasUtils.entityToReference(dbEntity.entity))
-    tblEntity.setRelationshipAttribute("sd", AtlasUtils.entityToReference(sdEntity.entity))
-
-    new AtlasEntityWithDependencies(tblEntity, Seq(dbEntity, sdEntity))
-  }
-
-  def hiveTableToEntitiesForAlterTable(
-      tblDefinition: CatalogTable,
-      cluster: String,
-      mockDbDefinition: Option[CatalogDatabase] = None): AtlasEntityWithDependencies = {
-    val tableEntity = hiveTableToEntity(tblDefinition, cluster, mockDbDefinition)
-    val deps = tableEntity.dependencies.map(_.entity)
-
-    val dbEntity = deps.filter(e => e.getTypeName.equals(HIVE_DB_TYPE_STRING)).head
-    val sdEntity = deps.filter(e => e.getTypeName.equals(HIVE_STORAGEDESC_TYPE_STRING)).head
-
-    // override attribute with reference - Atlas should already have these entities
-    tableEntity.entity.setRelationshipAttribute("db",
-      AtlasUtils.entityToReference(dbEntity, useGuid = false))
-    tableEntity.entity.setRelationshipAttribute("sd",
-      AtlasUtils.entityToReference(sdEntity, useGuid = false))
-
-    // drop all the dependencies since they're not necessary
-    AtlasEntityWithDependencies(tableEntity.entity)
-  }
-
-  // ================== Hive entities (Hive Warehouse Connector) =====================
-  val HWC_TABLE_TYPE_STRING = "hive_table"
-  val HWC_DB_TYPE_STRING = "hive_db"
-  val HWC_STORAGEDESC_TYPE_STRING = "hive_storagedesc"
-
-  def hwcTableUniqueAttribute(
-      cluster: String,
-      db: String,
-      tableName: String): String = {
-    s"${db.toLowerCase}.${tableName.toLowerCase}@$cluster"
+    hiveTableToReference(db, table, cluster)
   }
 
-  def hwcTableToEntity(
+  def hiveTableToReference(
       db: String,
       table: String,
-      cluster: String): AtlasEntityWithDependencies = {
-
-    val dbEntity = new AtlasEntity(HWC_DB_TYPE_STRING)
-    dbEntity.setAttribute("qualifiedName",
-      hiveDbUniqueAttribute(cluster, db.toLowerCase))
-    dbEntity.setAttribute("name", db.toLowerCase)
-
-    val sdEntity = new AtlasEntity(HWC_STORAGEDESC_TYPE_STRING)
-    sdEntity.setAttribute("qualifiedName",
-      hiveStorageDescUniqueAttribute(cluster, db, table))
-
-    val tblEntity = new AtlasEntity(HWC_TABLE_TYPE_STRING)
-    tblEntity.setAttribute("qualifiedName",
-      hiveTableUniqueAttribute(cluster, db, table))
-    tblEntity.setAttribute("name", table)
-    tblEntity.setAttribute("db",
-      AtlasUtils.entityToReference(dbEntity, useGuid = false))
-    tblEntity.setAttribute("sd",
-      AtlasUtils.entityToReference(sdEntity, useGuid = false))
-
-    AtlasEntityWithDependencies(tblEntity)
+      cluster: String): AtlasReferenceable = {
+    val qualifiedName = hiveTableUniqueAttribute(cluster, db, table /* , isTemporary = false */)
```

Comment:
is `isTemporary` used? If its not used now, can we remove it?. It seems the code generates a random string for `isTemporary ` and in that case how would it match with Atlas entitity Id?

Reply:
Yes actually `isTemporary` must be always `false`. We can just remove it explicitly, and look back if we encounter the needs. (I guess it will be just unsupported case since we can't create deterministic qualifiedName)

Reply:
ok to remove it.